### PR TITLE
Spacecraft sends notifications when certain events occur

### PR DESCRIPTION
**Pull request by CodeMartian**
#19 

**CHANGELOG** :memo:

* adds a SpacecraftEmitter that will send out messages depending on the following situations:
  -  max_speed: spacecraft attempts to accelerate faster than 5
  - min_speed: spacecraft attempts to decelerate slower than 1
  - wrong_trajectory: spacecraft traverses to the left or right farther than 5
  - moon: the spacecraft reaches the moon
  - lost: the spacesraft passes by the moon

**PRINTS(OPTIONAL)** :framed_picture:
**Spacecraft going too fast and going off course at the same time**
Spacecraft events

**Spacecraft going at the minimum speed, finding the moon, and passing the moon**
Spacecraft events 2

**LABELS**
* Enhancement
* Hacktoberfest

### CODE REVIEW

**Comment by CodeMartian** [comment]
Okay! @pyymenta I believe this PR is ready for your review. I did some fixin' up and polishing after I created the PR, but I believe it's in a good spot now. I'd love to hear your feedback!

**Comment by pyymenta** [comment]
@CodeMartian Thank you for your contribuition, I'll review it!

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
   "scripts": {
     "start": "node dist/main.js",
     "build": "tsc --build .tsconfig.json",
+    "test:dev": "jest --no-cache --watchAll",
```
Comment (pyymenta) [inline package.json]: Watch command, TDD MODE :on: 

:rocket: :rocket: :rocket:

File: src/main.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import CLI from './cli';
 import SpaceCraft from './spacecraft';
 import { Earth, Moon } from './locations';
+import SpacecraftEmitter from './spacecraftEmitter';
 
-const spacecraft = new SpaceCraft(Earth, Moon);
+const spacecraft = new SpaceCraft(Earth, Moon, new SpacecraftEmitter(process.stdout));
```
Comment (pyymenta) [inline src/main.ts]: I think you could declare the SpacecraftEmitter in a const and then, pass the instance to the SpaceCraft:

```suggestion
const spaceCraftEmitter = new SpacecraftEmitter(process.stdout);
const spacecraft = new SpaceCraft(Earth, Moon, spaceCraftEmitter);
```

File: src/spacecraftEmitter.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { EventEmitter } from 'events';
+import { event_messages } from './events_messages.json';
```
Comment (pyymenta) [inline src/spacecraftEmitter.ts:2]: Used the @danieljo09 list events :rocket: :rocket: :rocket: :rocket:

File: src/spacecraftEmitter.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export default class SpacecraftEmitter {
+	private eventEmitter: EventEmitter = new EventEmitter();
+	constructor(output: NodeJS.WriteStream) {
+		event_messages.forEach((message) => {
+			this.eventEmitter.on(message.id, () => {
+				output.write(` ${message.message} `);
+			});
+		});
```
Comment (pyymenta) [inline src/spacecraftEmitter.ts:16]: I believe it should be a method, to keep the constructor clean, with only a function call. What do you think?

File: src/spacecraftEmitter.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+	}
+
+	report(event: string) {
+		const event_message = event_messages.find((message) => {
+			return message.id === event;
+		});
+		if (event_message) {
```
Comment (pyymenta) [inline src/spacecraftEmitter.ts]: Could you add a space between these commands?

**Comment by pyymenta** [comment]
In general your implementation looks good to me, if you fix those points, I believe that will be ready to be merged.

One point of attention is about the identation, I think 4 spaces are not good. I'm thinking to changing it to 2. What do you think? No worry about it now, I'll create another issue for it.

File: package.json (hunk 1 above)
Comment (CodeMartian) [reply package.json]: 🥳

**Review by pyymenta — APPROVED** [review]
Very good @CodeMartian!! You did it! Could I merge in? Or you still working in it?

**Comment by CodeMartian** [comment]
go ahead and 🚀 🚀 🚀 🚀 🚀

**Comment by pyymenta** [comment]
> go ahead and

Awesome! Thank you!

**Pull request merged**